### ISSUE TITLE
Validation of incoming constellations

### ISSUE THREAD

**Issue by deternitydx** [snac-cooperative/snac#53]
In gitlab by @deternitydx on Aug 25, 2017, 16:15

Doug has found a way to insert a constellation with Resource Relations without associated Resource objects.  The Validation Engine and/or DBUtil needs to be updated to reject cases like this.  Did the Validation Engine actually make it into DBUtil's write methods?

**Comment by deternitydx** [comment]
In gitlab by @deternitydx on Sep 13, 2017, 16:40

In DBUtil, Tom had run validation, but never acted upon the result.  Specifically, a log message is printed, but the constellation is allowed to write:

```php
        if (!$ve->validateConstellation($cObj))
        {
            // problem
            $this->logDebug(sprintf("Error: Validation failed: %s", $ve->getErrors()));
        }
```

**Comment by deternitydx** [comment]
In gitlab by @deternitydx on Sep 13, 2017, 16:45

However!  The ServerExecutor *does* do validation before a write:

```php
            try {
                $validation = new \snac\server\validation\ValidationEngine();
                $hasOperationValidator = new \snac\server\validation\validators\HasOperationValidator();
                $validation->addValidator($hasOperationValidator);

                $success = $validation->validateConstellation($constellation);

            } catch (\snac\exceptions\SNACValidationException $e) {
                // If the Constellation has no changes, then don't do anything and allow the "update"
                // but don't do anything
                $response["constellation"] = $constellation->toArray();
                $response["result"] = "success";
                return $response;
            }
```

**Comment by deternitydx** [comment]
Fixed in commit 25acb84.  Any constellation write now goes through the ResourceValidator that will check that any ResourceRelation has a Resource object with an ID and Version.

**Comment by deternitydx** [comment]
Resource Validator now on master.
